### DISCUSSION TITLE
Getting associated error codes and messages?

### DISCUSSION THREAD

**Discussion by bytelabsco** [bartoszlenar/Validot#35] [Q&A]
I think this may be related to #34, but just wanted to confirm my understanding.  If I have multiple rules defined for a property with messages and codes (a simplified example from #34)

```
Specification spec = s => s
.Rule(x => false).WithMessage("Message 11").WithExtraCode("Code 12")
.Rule(x => false).WithMessage("Message 21").WithExtraCode("Code 22")
```
upon validation failing for both rules, is there a way to get the message/code groupings?  What I mean by this is a way to get that "Message 11" and "Code 12" are related, and "Message 21" and "Code 22" are related?  It seems at best I can use the Path as the key in the MessageMap and CodeMap, but that doesn't tie those items together.

**Comment by bartoszlenar** [comment]
Hi @bytelabsco , sorry for the late answer.

You're right. #34 is related. 
Validot doesn't hold any extra information about messages and their relation or occurrence in the specification. 

Can you evaluate using any specific example when that scenario could be useful?
